### PR TITLE
Feat/137 verbose variable

### PR DESCRIPTION
**Pull request by travela**
We used tqdm library for progress bars.

The PropertyInferenceAttack class now takes an optional argument verbose from {0, 1, 2}. With 0 = silent, 2 = most verbose.
The trainer function in utils/trainer.py also takes a verbose argument.

Solves #137

### CODE REVIEW

File: privacy_evaluator/utils/model_utils.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            for layer in layers:
+                if hasattr(layer, "reset_parameters"):
+                    layer.reset_parameters()
+        elif hasattr(layers, "reset_parameters"):
+            layers.reset_parameters()
+    return model
+
+
+def _copy_and_reset_tf_model(target_model: keras.Model) -> keras.Model:
+    target_model.save("./target_model")
```
Comment (TobiasSchalau) [inline privacy_evaluator/utils/model_utils.py:35]: The relative path is from where the function is executed right? Maybe it's to unspecific?

File: privacy_evaluator/utils/trainer.py [hunk 2]
```diff
@@ -146,13 +164,14 @@ def _trainer_torch(
     class_encoding = {class_id: i for i, (class_id, _) in enumerate(size_dict.items())}
 
     # start training
-    for _ in range(num_epochs):
+    if verbose==2:
+        print("Training PyTorch model in ", num_epochs, "epochs.")
```
Comment (TobiasSchalau) [inline privacy_evaluator/utils/trainer.py:168]: missing space in front of "epochs." ?

File: tests/attacks/property_inference_test.py [hunk 3]
```diff
@@ -9,18 +9,19 @@
...
-    num_elements_per_classes = {0: 1000, 1: 1000}
+    num_elements_per_classes = {0: 500, 1: 500}
     num_classes = len(num_elements_per_classes)
 
     train_set = new_dataset_from_size_dict(train_dataset, num_elements_per_classes)
 
     model = ConvNet(num_classes, input_shape)
-    trainer(train_set, num_elements_per_classes, model)
+    print("Start training target model ...\n")
+    trainer(train_set, num_elements_per_classes, model, num_epochs=2, verbose=2)
```
Comment (TobiasSchalau) [inline tests/attacks/property_inference_test.py:21]: How I understood the test it's an intern function for pytest. So in my point of view a verbose of 2 or 1 isn't necessary

File: privacy_evaluator/utils/trainer.py [hunk 4]
```diff
@@ -86,7 +101,9 @@ def _trainer_tf(
     class_encoding = {class_id: i for i, (class_id, _) in enumerate(size_dict.items())}
 
     # start training
-    for _ in range(num_epochs):
+    if verbose==2:
+        print("Training TensorFlow model in ", num_epochs, "epochs.")
```
Comment (TobiasSchalau) [inline privacy_evaluator/utils/trainer.py:105]: missing space in front of "epochs."?

File: privacy_evaluator/attacks/property_inference_attack.py [hunk 5]
```diff
@@ -338,17 +344,24 @@ def attack(self):
...
+
         # set ratio and size for unbalanced data sets
         size_set = 1000
 
         # balanced ratio
         num_elements = int(round(size_set / 2))
         neg_property_num_elements_per_class = {0: num_elements, 1: num_elements}
 
+        if(self.verbose>0):
+            print("Creating set of balanced shadow classifiers ... ")
```
Comment (TobiasSchalau) [inline privacy_evaluator/attacks/property_inference_attack.py:364]: A more concrete message would include the number of balanced shadow classifiers

**Review by TobiasSchalau — COMMENTED** [review]
I put some commments on the code

File: privacy_evaluator/utils/model_utils.py (hunk 1 above)
Comment (travela) [reply privacy_evaluator/utils/model_utils.py:35]: This was kind of a workaround for the tensorflow case. We delete the directory again two lines later, so it shouldn't matter where exactly it is

File: privacy_evaluator/utils/trainer.py (hunk 2 above)
Comment (travela) [reply privacy_evaluator/utils/trainer.py:168]: There's space after "in". But "print(a, b)" automatically adds a space between a and b (so technically there's a space too much in the above code :D)

File: privacy_evaluator/attacks/property_inference_attack.py (hunk 5 above)
Comment (travela) [reply privacy_evaluator/attacks/property_inference_attack.py:428]: Good point. I changed that

File: tests/attacks/property_inference_test.py (hunk 3 above)
Comment (travela) [reply tests/attacks/property_inference_test.py:21]: Will delete it then

**Pull request merged**